src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -130,13 +121,19 @@ public void EndEditSession(out ImmutableArray<DocumentId> documentsToReanalyze)
             // clear all reported rude edits:
             documentsToReanalyze = session.GetDocumentsWithReportedDiagnostics();
 
-            _debuggingSessionTelemetry.LogEditSession(_editSessionTelemetry.GetDataAndClear());
+            // TODO: report a separate telemetry data for hot reload sessions to preserve the semantics of the current telemetry data
```

Comment:
i woudl love threading asserts on all these methods (i.e. AssertIsForeground/AssertIsBackground/ThisCanBeCalledOnAnyThread.  It makes reading and understanding potential threading/sync/consistency issues more simply.

Reply:
Why? There is no thread affinity. By default we don't put attributes on methods that don't have affinity.